src/js/scope_start.js

```diff
@@ -1,5 +1,8 @@
 
 function closure ( target, options ){
+	// clone the options because we are extending them
+	// so we should not modify the user provided object
+	options = clone(options);
```

Comment:
I get why you'd be hesitant to mutate the users options object. I'd rather solve this by copying the validated options in the `testOptions` method, instead of including the entire copy function.